### PR TITLE
fix: support files with spaces in `generateImageID`

### PR DESCRIPTION
**Pull request by pzerelles**
- **Quick Checklist**

* [x] I have read [the contributing guidelines](../CONTRIBUTING.md)
* [ ] I have written new tests, as applicable (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] I have added a changeset, if applicable

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

- **What is the new behavior (if this is a feature change)?**

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this
  PR?)

- **Other information**:

### CODE REVIEW

**Comment by benmccann** [comment]
This is included in the change to https://github.com/JonasKruckenberg/imagetools/pull/670, so we only need one

**Comment by benmccann** [comment]
@pzerelles is this PR still necessary if https://github.com/vitejs/vite/pull/15311 is merged?

**Comment by pzerelles** [comment]
> @pzerelles is this PR still necessary if [vitejs/vite#15311](https://github.com/vitejs/vite/pull/15311) is merged?

Yes

File: .changeset/angry-tigers-tickle.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'vite-imagetools': patch
+---
+
+fix: generateImageID failing for files with spaces
```
Comment (benmccann) [inline .changeset/angry-tigers-tickle.md:5]:
```suggestion
fix: make `generateImageID` work for files with spaces
```

**Review by benmccann — APPROVED** [review]

**Pull request merged**